### PR TITLE
Restructure Baselines docs

### PR DESCRIPTION
**Pull request by charlesbvll**
### Description

The Baselines docs do not currently follow the Diataxis framework.

### Related issues/PRs

N/A

### Explanation

Structure the docs into:

- Tutorials
- Guides
- Explanation
- References

### Checklist

- [x] Implement proposed change
- [x] Make CI checks pass
- [x] Ping maintainers on [Slack](https://flower.dev/join-slack/) (channel `#contributions`)

### Any other comments?

N/A

### CODE REVIEW

File: baselines/doc/source/index.rst [hunk 1]
```diff
@@ -25,6 +25,8 @@ Flower Baselines are a collection of organised scripts used to reproduce results
    :maxdepth: 1
    :caption: Flower Baselines
    
-   using-baselines
-   contributing-baselines
+   tutorials
+   guides
+   reference
+   explanations
```
Comment (tanertopal) [inline baselines/doc/source/index.rst:31]: @charlesbvll is it possible to start in expanded mode?

Comment (charlesbvll) [reply baselines/doc/source/index.rst:31]: @tanertopal No, unfortunately, this is something that Daniel asked me as well but it is not possible (except if we fork the Sphinx project or the Furo theme)

Comment (tanertopal) [reply baselines/doc/source/index.rst:31]: @charlesbvll do you know what would need to be changed in Furo? Afaik it's easy to override template files.

Comment (charlesbvll) [reply baselines/doc/source/index.rst:31]: @tanertopal This is the approach I attempted but when modifying the template file I could expand it but then the collapsing functionality didn't work anymore

**Review by tanertopal — APPROVED** [review]

**Pull request merged**